compiler/src/dotty/tools/dotc/transform/BetaReduce.scala

```diff
@@ -128,7 +128,10 @@ object BetaReduce:
             ref.symbol
           case _ =>
             val flags = Synthetic | (param.symbol.flags & Erased)
-            val tpe = if arg.tpe.dealias.isInstanceOf[ConstantType] then arg.tpe.dealias else arg.tpe.widen
+            val tpe =
+              if arg.tpe.isBottomType then param.tpe.widenTermRefExpr
```

Comment:
I'm not sure why we use `widenTermRefExpr` here (and not `widen` for example), just curious.

Reply:
To keep the type precise. For example, if we have a singleton `String` type we do not want to widen it to `String`. For example `((x: "foo") => x: "foo").apply(???)` should become `val x: "foo" = ???; x: "foo"` and not `val x: String = ???; x: "foo"`.